server/automute_channel.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/plugin"
+
+	"github.com/pkg/errors"
+)
+
+func (p *Plugin) UserHasJoinedChannel(c *plugin.Context, channelMember *model.ChannelMember, actor *model.User) {
+	_, _ = p.updateAutomutingOnUserJoinedChannel(c, channelMember.UserId, channelMember.ChannelId)
+}
+
+func (p *Plugin) updateAutomutingOnUserJoinedChannel(c *plugin.Context, userID string, channelID string) (bool, error) {
+	if automuteEnabled := p.getAutomuteIsEnabledForUser(userID); !automuteEnabled {
+		return false, nil
+	}
+
+	if canAutomute, err := p.isChannelLinked(channelID); err != nil {
+		p.API.LogError(
+			"Unable to check if channel is linked to update automuting when a user has joined the channel",
+			"UserID", userID,
+			"ChannelID", channelID,
+			"Error", err.Error(),
```

Comment:
A recent change, but let's use snake case here:
```suggestion
			"user_id", userID,
			"channel_id", channelID,
			"error", err.Error(),
```